dockstore-webservice/src/test/java/io/dockstore/webservice/helpers/SourceCodeRepoFactoryTest.java

```diff
@@ -27,19 +27,38 @@
 
     @Test
     public void parseGitUrl() {
+        // https://stackoverflow.com/questions/59081778/rules-for-special-characters-in-github-repository-name
         // test format 1
         final Map<String, String> stringStringMap1 = SourceCodeRepoFactory.parseGitUrl("git@github.com:dockstore/dockstore-ui.git");
         Assert.assertNotNull(stringStringMap1);
         Assert.assertTrue("values not found",
-                stringStringMap1.containsKey("Source") && stringStringMap1.containsKey("Username") && stringStringMap1
-                        .containsKey("Repository"));
+                stringStringMap1.containsKey("Source") && stringStringMap1.get("Source").equals("github.com")
```

Comment:
The `containsKey` are redundant, although they don't hurt.

But I would suggest: 

```
Assert.assertEquals("github.com", stringStringMap1.get("Source");
Assert.assertEquals("dockstore", stringStringMap1.get("Username");
etc.
```

That way you'll get a better error message if it fails.

Reply:
Also, it's only a test, but you have the strings "Username", "Source", and "Repository" 8 times each (it would still be 4 times if you get rid of the `containsKey`) -- good candidates for variables.

Reply:
done